pkg/plugins/backendplugin/grpcplugin/client_proto.go

```diff
@@ -36,14 +36,17 @@ type ProtoClient interface {
 	Logger() log.Logger
 	Start(context.Context) error
 	Stop(context.Context) error
-	Running(context.Context) bool
 }
 
 type protoClient struct {
 	plugin        *grpcPlugin
 	pluginVersion string
 	pluginJSON    plugins.JSONData
 
+	startInit    bool
+	startSuccess bool
+	startFail    bool
```

Comment:
It looks like these can only be in one "state" at the same time. Would it make sense to change it to something like this?

```go
type pluginState int

const (
	pluginStateStopped pluginState = iota
	pluginStateInit
	pluginStateSuccess
	pluginStateFail
)
```

```suggestion
	state pluginState
```